Sources/MongoSwift/BSON/BSONValue.swift

```diff
@@ -387,6 +394,14 @@ public struct Binary: BSONValue, Equatable, Codable {
     }
 }
 
+// An extension of `BSONUndefined` to add capability to be hashed
+extension Binary: Hashable {
+    public func hash(into hasher: inout Hasher) {
+        hasher.combine(data)
```

Comment:
In cases where all the members of a struct / class conform to `Hashable`, the compiler actually "synthesizes" the conformance for us. For example, in this case you can simply update the declaration of `Binary` to `Binary: BSONValue, Equatable, Hashable, Codable` and remove this hash function and it be generated for you at compile time. You may actually omit `Equatable` in the declaration I think since `Hashable` implies it.

The same thing applies to `Equatable`. You'll notice there's no `==` defined on `Binary`, we just get it for free since `data` and `subtype` conform to `Equatable`.

This synthesization applies to any type whose `hash` function is just combining all it's fields (e.g. `Timestamp`)